Cargo.toml

```
@@ -24,10 +24,13 @@ hyper = { version = "1", features = ["full"] }
 hyper-rustls = { version = "0.27", default-features = false, features = ["webpki-roots", "ring", "http1"] }
 hyper-util = { version = "0.1", features = ["client", "client-legacy", "tokio", "http2"] }
 opentelemetry = { version = "0.28", optional = true }
-quick-error = "2"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 slog-scope = "4"
 smart-default = "0.7"
+thiserror = "2.0.12"
```

Nit: We try to limit to the most significant part of the version as it prevents confusion for newer rust developers who may think this is being pinned to a particular version
```suggestion
thiserror = "2"
```